R/rmd_to_md.R

```diff
@@ -0,0 +1,94 @@
+#' Convert external R Markdown to Markdown
+#'
+#' Converts an R Markdown (.Rmd) file to a Markdown (.md) file. Both local and
+#' remote .Rmd files can be handled. The date this function is called upon is
+#' added to the beginning of the Markdown file.
+#'
+#' @param rmd_file Path to the R Markdown file, either a local path or a URL.
+#' @param md_dir Path to local directory to Markdown file to.
+#' @param fig_dir Path to local directory to save figures to.
+#' @param fig_url_dir URL path that will be used to link to the figures in the
+#' markdown output.
+#' @param order Order of the article in the menu.
+#'
+#' @return Markdown file and figures written do disk.
+#' @export
+#'
+#' @examples
+#' rmd_file <- "https://raw.githubusercontent.com/b-cubed-eu/gcube/refs/heads/main/vignettes/articles/occurrence-process.Rmd"
+#'
+#' md_dir <- file.path("output", "src", "content", "docs", "software", "gcube")
+#' fig_dir <- file.path("output", "public", "software", "gcube")
+#' fig_url_dir <- "/software/gcube/"
+#' rmd_to_md(rmd_file, md_dir, fig_dir, fig_url_dir, order = 1)
+#'
+#' # Clean up (don't do this if you want to keep your files)
+#' unlink("output", recursive = TRUE)
+rmd_to_md <- function(rmd_file, md_dir, fig_dir, fig_url_dir, order) {
+  md_name <- gsub(".Rmd$", "", basename(rmd_file))
+
+  # Set input
+  if (grepl("^http", rmd_file)) {
+    # Correct mixed slash and backslash in file path (in Windows tempdir() uses
+    # double backslashes as separator while file.path() uses regular slashes.)
+    tempdir <- gsub("\\\\", "/", tempdir())
+    if (dir.exists(tempdir)) {
```

Comment:
I'm a bit confused as to what is happening here. Do you have an example of what values for `tempdir()` would trigger the `if(){}` clause?